server/ServerFSM.cpp

```diff
@@ -1321,6 +1370,11 @@ sc::result WaitingForTurnEndIdle::react(const SaveGameRequest& msg) {
     return transit<WaitingForSaveData>();
 }
 
+sc::result WaitingForTurnEndIdle::react(const Error& msg) {
```

Comment:
HandleErrorMessage() doesn't return if the error is fatal.  I've added a comment to its definition.